src/components/networkContext/NetworkContext.tsx

```diff
@@ -0,0 +1,79 @@
+import { Repo } from "@automerge/automerge-repo";
+import { BrowserWebSocketClientAdapter } from "@automerge/automerge-repo-network-websocket";
+import { RepoContext } from "@automerge/automerge-repo-react-hooks";
+import { IndexedDBStorageAdapter } from "@automerge/automerge-repo-storage-indexeddb";
+import { Loader } from "@mantine/core";
+import { createContext, useCallback, useEffect, useState } from "react";
+import { TRepoContextValue, TRepoProviderProps } from "./types";
+
+const LOCAL_STORAGE_NETWORK_URL_KEY = "ytbNetworkUrl";
+
+const NetworkContext = createContext<TRepoContextValue>({
+  setNetworkUrl: () => null,
+});
+
+export const NetworkProvider: React.FC<TRepoProviderProps> = ({ children }) => {
+  const [repo, setRepo] = useState<Repo>();
```

Comment:
I would probably use a `useRef` hook instead of an `useState` for creating the instances.